### PR TITLE
Add test runner for various console attachment configurations

### PR DESCRIPTION
**Pull request by ghale**
See https://github.com/gradle/gradle-native/issues/668.  This simplifies our console integration tests so that we don't have to have a different subclass for each console attachment configuration.

### CODE REVIEW

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/console/AbstractConsoleGroupedTaskFunctionalTest.groovy [hunk 1]
```diff
@@ -19,36 +19,47 @@ package org.gradle.integtests.fixtures.console
...
  */
+@RunWith(ConsoleAttachmentTestRunner.class)
 abstract class AbstractConsoleGroupedTaskFunctionalTest extends AbstractIntegrationSpec implements RichConsoleStyling {
+    static ConsoleAttachment consoleAttachment
+
     protected boolean consoleAttached
     protected boolean stderrAttached
 
     def setup() {
+        // TODO - rich/verbose consoles currently exhibit slightly different behavior than these tests expect when no console is attached
```
[the remark this answers is not in the document]
Comment (ghale) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/console/AbstractConsoleGroupedTaskFunctionalTest.groovy]: Yep.  I'm going to work on that story next, so I don't expect this TODO to be here long.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/console/AbstractConsoleGroupedTaskFunctionalTest.groovy [hunk 2]
```diff
@@ -19,36 +19,47 @@ package org.gradle.integtests.fixtures.console
...
+        Assume.assumeFalse(consoleAttachment == ConsoleAttachment.NOT_ATTACHED && (consoleType == ConsoleOutput.Rich || consoleType == ConsoleOutput.Verbose))
+
+        switch(consoleAttachment) {
+            case ConsoleAttachment.NOT_ATTACHED:
+                consoleAttached = false
+                stderrAttached = false
```
[the remark this answers is not in the document]
Comment (ghale) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/console/AbstractConsoleGroupedTaskFunctionalTest.groovy]: Good idea.  Done.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/console/AbstractConsoleGroupedTaskFunctionalTest.groovy [hunk 3]
```diff
@@ -19,36 +19,38 @@ package org.gradle.integtests.fixtures.console
...
+        Assume.assumeFalse(consoleAttachment == ConsoleAttachment.NOT_ATTACHED && (consoleType == ConsoleOutput.Rich || consoleType == ConsoleOutput.Verbose))
+
+        switch(consoleAttachment) {
+            case ConsoleAttachment.NOT_ATTACHED:
+                break
+            case ConsoleAttachment.ATTACHED:
+                executer.withTestConsoleAttached()
+                break
+            case ConsoleAttachment.ATTACHED_STDOUT_ONLY:
+                executer.withTestConsoleAttachedToStdoutOnly()
```
Comment (adammurdoch) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/console/AbstractConsoleGroupedTaskFunctionalTest.groovy]: would be simpler if there were a `withConsoleAttachment()` on `GradleExecuter` instead.

Comment (adammurdoch) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/console/AbstractConsoleGroupedTaskFunctionalTest.groovy]: ah, I see this is in the next PR. Nevermind then.

**Pull request merged**